app/controllers/courses/modern_languages_controller.rb

```diff
@@ -0,0 +1,73 @@
+module Courses
+  class ModernLanguagesController < ApplicationController
+    include CourseBasicDetailConcern
+    decorates_assigned :course
+    before_action :build_course, only: %i[edit update]
+
+    def edit
+      return unless @course.meta[:edit_options][:modern_languages].nil?
+
+      redirect_to(
+        details_provider_recruitment_cycle_course_path(
+          @course.provider_code,
+          @course.recruitment_cycle_year,
+          @course.course_code,
+        ),
+      )
+    end
+
+    def update
+      if @course.update(subjects: old_non_language_subjects + new_language_subjects)
```

Comment:
I'm struggling to decipher the intent behind `old_non_language_subjects + new_language_subjects`

Reply:
explanation:

We are modifying the list of *language* subjects, but leaving the other types of subjects intact.
Because it's a replacemend of the whole subject list we need to reconstruct the combined result.